common/plugins/eu.esdihumboldt.hale.common.instance.graph/src/eu/esdihumboldt/hale/common/instance/graph/reference/ReferenceGraph.java

```diff
@@ -235,24 +241,24 @@ public Boolean compute(LoopBundle<Vertex> loop) {
 					if (ref != null) {
 						result.add(ref);
 					}
-					else {
+					else if (!vtx.getId().equals(associated.getId())) {
 						Iterable<Vertex> referers = associated.getVertices(Direction.IN);
 						Set<String> ids = new HashSet<>();
-						if (referers != null) {
-							for (Vertex referer : referers) {
-								Object ident = referer.getId();
-								if (ident != null) {
-									ids.add(ident.toString());
-								}
+
```

Comment:
Any particular reason to remove the null check?

Reply:
I haven't, it's just below.

Reply:
No, `referers` is no longer checked for `null`.
Did you check the API and it is guaranteed to never be `null`?

Reply:
The `associated.getVertices(Direction.IN)` will return an empty iterable if there are no vertices.
I actually wanted to used this 
```
Set<String> ids = new HashSet<>();
for (Vertex referer : associated.getVertices(Direction.IN)) {
    Object ident = referer.getId();
    if (ident != null) {
        ids.add(ident.toString());
    }
}
```